packages/cli/src/ui/App.tsx

```diff
@@ -175,45 +105,74 @@ const App = ({ config, settings, startupWarnings = [], version }: AppProps) => {
   const isFocused = useFocus();
   useBracketedPaste();
   const [updateInfo, setUpdateInfo] = useState<UpdateInfo | null>(null);
-  const [isUpdating, setIsUpdating] = useState(false);
   const { stdout } = useStdout();
   const nightly = version.includes('nightly');
   const { history, addItem, clearItems, loadHistory } = useHistory();
 
   useEffect(() => {
-    checkForUpdates().then(async (info) => {
-      if (!info) {
-        return;
-      }
-
-      const isNpx = process.env.npm_execpath?.includes('npx');
-      const isGit = await isGitRepository();
-
-      if (isGit || isNpx || settings.merged.disableAutoUpdate) {
-        const notificationInfo = { ...info };
-        if (!(isGit || isNpx)) {
-          // Add manual install instructions for regular installs with auto-update disabled
-          notificationInfo.message += `\nRun npm install -g ${info.update.name} to install`;
-        }
-
-        setUpdateInfo(notificationInfo);
-
-        setTimeout(() => {
+    let successfullyInstalled = false;
+    const handleUpdateRecieved = (info: UpdateInfo) => {
+      setUpdateInfo(info);
+      const savedMessage = info.message;
+      setTimeout((_: UpdateInfo) => {
+        if (!successfullyInstalled) {
           addItem(
             {
               type: MessageType.INFO,
-              text: notificationInfo.message,
+              text: savedMessage,
             },
             Date.now(),
           );
-          setUpdateInfo(null);
-        }, 60000);
-        return;
-      }
+        }
+        setUpdateInfo(null);
+      }, 60000);
```

Comment:
why do we need this 60 second timeout? Seems like we could just be adding the item right at the line where we set 
`successfullyInstalled = true`

Reply:
Do you mean immediately adding them item and not showing the notification? the timeout is there because this also covers the case where the user doesn't want to autoupdate or they can't so it will show the notification for a minute and then add it to history